cadc-inventory/src/main/java/org/opencadc/inventory/InventoryUtil.java

```diff
@@ -141,22 +145,66 @@ public static String computeArtifactFilename(URI uri) {
             throw new IllegalStateException("CONFIG: " + cnameProp + " not set");
         }
         try {
-            Class c = Class.forName(cname);
-            return (T) c.newInstance();
+            Class<?> c = Class.forName(cname);
+            return (T) c.getDeclaredConstructor().newInstance();
+        } catch (InvocationTargetException ex) {
+            throw new IllegalStateException("CONFIG: " + cnameProp + " implementation crashed during creation.", ex);
         } catch (ClassNotFoundException ex) {
             throw new IllegalStateException("CONFIG: " + cnameProp + " implementation not found in classpath: " + cname, ex);
-        } catch (InstantiationException ex) {
+        } catch (InstantiationException | NoSuchMethodException ex) {
             throw new IllegalStateException("CONFIG: " + cnameProp + " implementation " + cname + " does not have a no-arg constructor", ex);
         } catch (IllegalAccessException ex) {
             throw new IllegalStateException("CONFIG: failed to instantiate " + cname, ex);
         }
     }
-    
+
+    /**
+     * Load and instantiate an instance of the specified Java interface.  This method uses
+     * the fully-qualified class name as a system property key; the value of the system property
+     * is the fully qualified class name of an implementation of that interface.
+     *
+     * <p>This overloaded method will use a series of constructor arguments to build a new instance.  It assumes that
+     * the requested Class contains a constructor with the given arguments.
+     *
+     * @param <T>             Class type of the instantiated class
+     * @param implementationClassName   Class name to create
+     * @param constructorArgs The constructor arguments
+     * @return configured implementation of the interface
+     *
+     * @throws IllegalStateException if an instance cannot be created
+     */
+    public static <T> T loadPlugin(final String implementationClassName, final Object... constructorArgs)
```

Comment:
This is actually really cool, although takes some getting used to.  In later versions of Java this was improved, but for Java 8 this is as good as it can be, I think.  The idea is that it takes on the type of the variable it's being assigned to.  If they don't match, there is a `ClassCastException`.

Reply:
I can't decide if it is cool or a land mine waiting to bite someone. You have a totally innocent line of code like:
```
Foo foo = loadPlugin(...);
```
that can throw a ClassCastException right there. I'll toy around a little at some point and see if I can make it safer, but for the caller should catch ClassCastException and wrap it in IllegalArgumentException.

Reply:
I've eliminated this for DAO creation.  I think this way of creating objects can cause issues in the end.  All in all, one _could_ create their entire model with this style of operation, but it ignores the contracts that were created in honest when the model was designed.  Not to mention all of the `ClassCastException` handling.